i3-settings-qubes/qubes-i3status

```diff
@@ -45,26 +45,49 @@ status_time() {
 }
 
 status_bat() {
-    local accum_now=0;
-    local accum_full=0;
+    local accum_now_mWh=0 # current battery power in mWh
+    local accum_full_mWh=0 # full battery power in mWh
 
-    for battery in /sys/class/power_supply/BAT*; do
-        accum_now=$((accum_now + $(cat $battery/energy_now)))
-        accum_full=$((accum_full + $(cat $battery/energy_full)))
+    local batteries # batteries connected to the system
+    mapfile -t batteries < <(shopt -s nullglob; for f in /sys/class/power_supply/BAT*; do echo "$f"; done)
+    for battery in "${batteries[@]}"; do
+        if [ -f "${battery}"/energy_now ]; then
+            accum_now_mWh=$((accum_now_mWh + $(cat "${battery}"/energy_now)))
+            accum_full_mWh=$((accum_full_mWh + $(cat "${battery}"/energy_full)))
+        elif [ -f $battery/charge_now ]; then
+            # charge is given in mAh, convert to mWh
+            local voltage=$(cat "${battery}"/voltage_now)
+            local now_mWh=$(((voltage / 1000) *  $(cat "${battery}"/charge_now) / 1000))
+            local full_mWh=$(((voltage / 1000) *  $(cat "${battery}"/charge_full) / 1000))
```

Comment:
Could save some cat execs with $(<file) if you wanted to. You're already using #!/bin/bash instead of #!/bin/sh, so posix-sh compatibility is not a concern. IMO it's nice for periodic things like this to be light-weight.
Doesn't really matter though.